Web/WebServices2/src/com/metacube/webservices/EmployeeService.java

```diff
@@ -0,0 +1,225 @@
+			return "<insertEmployee>" + output + "</insertEmployee>";
+		} else {// if not inserted successfully
+			output = "@Produces(\"application/xml\") Output: \n"
+					+ "Employee not Inserted";
+			return "<insertEmployee>" + output + "</insertEmployee>";
+		}
+
+	}
+
+	/**
+	 * function to show employee by id
+	 * 
+	 * @param id
+	 *            :id of employee
+	 * @return String of employee of this id as output
+	 * @throws SQLException
+	 */
+	@POST
+	@Produces("application/xml")
+	@Path("{id}")
+	public String selectEmployeeById(@PathParam("id") int id)
+			throws SQLException {
+		// getting connection
+		Connection connection = ConnectionFactory.getConnection();
+		// use employee database
+		EmployeeDao.useEmployeeDatabase(connection);
+		String output = "";
+		// get employee based on id
+		Employee employeeById = EmployeeDao.selectListOfEmployeeById(
+				connection, id);
+
+		if (employeeById != null) {// if employee of this id available then
+			output = "@Produces(\"application/xml\") Output: \n"
+					+ "Employee Found";
+			return "<Employee>" + output + "\n" + "<employeeId>"
+					+ employeeById.getId() + "</employeeId>" + "\n"
+					+ "<employeeName>" + employeeById.getName()
+					+ "</employeeName>" + "\n" + "<employeeSalary>"
+					+ employeeById.getSalary() + "</employeeSalary>" + "\n"
+					+ "<employeeDept>" + employeeById.getDepartment()
+					+ "</employeeDept>" + "</Employee>";
+		} else {// if this employee not available
+
+			output = "@Produces(\"application/xml\") Output: \n"
+					+ "Employee not found";
+			return "<EmployeeById>" + output + "</EmployeeById>";
+		}
+	}
+
+	/**
+	 * 
+	 * function to return list of employee by name
+	 * 
+	 * @param name
+	 *            :name of employee
+	 * @return :string of employee of this name
+	 * @throws SQLException
+	 */
+	@POST
+	@Produces("application/xml")
+	@Path("name/{name}")
+	public String selectEmployeeByName(@PathParam("name") String name)
+			throws SQLException {
+		String output = "";
+		List<Employee> employeeList = null;
+		// getting connection
+		Connection connection = ConnectionFactory.getConnection();
+		// using employee database
+		EmployeeDao.useEmployeeDatabase(connection);
+		// getting employee based on employee name
+		employeeList = EmployeeDao.selectListOfEmployeeByName(connection, name);
+		if (!employeeList.isEmpty()) {// if employee list is not empty
+
+			Iterator<Employee> iterator = employeeList.iterator();
+			while (iterator.hasNext()) {// iterating employee
+				Employee employee = iterator.next();
+
+				output += "<Employee>" + "\n" + "<employeeId>"
+						+ employee.getId() + "</employeeId>" + "\n"
+						+ "<employeeName>" + employee.getName()
+						+ "</employeeName>" + "\n" + "<employeeSalary>"
+						+ employee.getSalary() + "</employeeSalary>" + "\n"
+						+ "<employeeDept>" + employee.getDepartment()
+						+ "</employeeDept>" + "</Employee>";
+			}
+			return "<EmployeeById>" + output + "</EmployeeById>";
```

Comment:
ACTION: Why we are wrapping output within EmployeeById tag.? it should be EmployeeByName.